packages/flutter_tools/test/general.shard/cmake_test.dart

```diff
@@ -0,0 +1,158 @@
+  });
+
+  testUsingContext('parses executable name from cmake file', () async {
+    final FlutterProject project = FlutterProject.fromDirectoryTest(fileSystem.currentDirectory);
+    final CmakeBasedProject cmakeProject = _FakeProject.fromFlutter(project);
+
+    cmakeProject.cmakeFile
+      ..createSync(recursive: true)
+      ..writeAsStringSync('set(BINARY_NAME "hello")');
+
+    final String name = getCmakeExecutableName(cmakeProject);
+
+    expect(name, 'hello');
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => processManager,
+  });
+
+  testUsingContext('defaults executable name to null if cmake config does not exist', () async {
+    final FlutterProject project = FlutterProject.fromDirectoryTest(fileSystem.currentDirectory);
+    final CmakeBasedProject cmakeProject = _FakeProject.fromFlutter(project);
+
+    final String name = getCmakeExecutableName(cmakeProject);
+
+    expect(name, isNull);
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => processManager,
+  });
+
+  testUsingContext('generates config', () async {
+    final FlutterProject project = FlutterProject.fromDirectoryTest(fileSystem.currentDirectory);
+    final CmakeBasedProject cmakeProject = _FakeProject.fromFlutter(project);
+    final Map<String, String> environment = <String, String>{};
+
+    writeGeneratedCmakeConfig(
+      _kTestFlutterRoot,
+      cmakeProject,
+      environment,
+    );
+
+    final File cmakeConfig = cmakeProject.generatedCmakeConfigFile;
+
+    expect(cmakeConfig, exists);
+
+    final List<String> configLines = cmakeConfig.readAsLinesSync();
+
+    expect(configLines, containsAll(<String>[
+      '# Generated code do not commit.',
+      r'file(TO_CMAKE_PATH "/flutter" FLUTTER_ROOT)',
+      r'file(TO_CMAKE_PATH "/" PROJECT_DIR)',
+
+      '# Environment variables to pass to tool_backend.sh',
+      'list(APPEND FLUTTER_TOOL_ENVIRONMENT',
+      r'  "FLUTTER_ROOT=/flutter"',
+      r'  "PROJECT_DIR=/"',
+      ')',
+    ]));
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => processManager,
+  });
+
+  testUsingContext('config escapes backslashes', () async {
+    fileSystem = MemoryFileSystem.test(style: FileSystemStyle.windows);
+
+    final FlutterProject project = FlutterProject.fromDirectoryTest(fileSystem.currentDirectory);
+    final CmakeBasedProject cmakeProject = _FakeProject.fromFlutter(project);
+
+    final Map<String, String> environment = <String, String>{
+      'TEST': r'hello\world',
+    };
+
+    writeGeneratedCmakeConfig(
+      _kTestWindowsFlutterRoot,
+      cmakeProject,
+      environment,
+    );
+
+    final File cmakeConfig = cmakeProject.generatedCmakeConfigFile;
+
+    expect(cmakeConfig, exists);
+
+    final List<String> configLines = cmakeConfig.readAsLinesSync();
+
+    expect(configLines, containsAll(<String>[
+      '# Generated code do not commit.',
```

Comment:
I'd make all of the strings in this list raw, just for consistency, because someone might add a backslash later and not notice that the string wasn't raw (well, right away, at least).

```suggestion
      r'# Generated code do not commit.',
```

Reply:
That's a good point, updated!